src/main/java/com/shapesecurity/shift/es2016/parser/GenericParser.java

```diff
@@ -1346,6 +1345,11 @@ protected boolean lookaheadAssignmentExpression() {
 
         if (this.lookahead.type != TokenType.EXP) {
             return left;
+        } else if (left.isLeft()) {
+            Expression expression = left.left().fromJust();
+            if (expression instanceof UnaryExpression) {
+                throw this.createUnexpected(this.lookahead);
```

Comment:
This looks to me like it would reject `(-1) ** 2`. Do we have a test covering that case? If not we should add one.

Reply:
Test added, passing